### PR TITLE
✨ Fix missing item `id` when attempting to index `custom_item_data`

### PR DESCRIPTION
**Pull request by MulverineX**

### CODE REVIEW

File: packages/mcdoc/src/runtime/checker/index.ts [hunk 1]
```diff
@@ -877,18 +877,14 @@ function simplifyIndexed<T>(
...
 		return { kind: 'union', members: [] }
 	}
 	let values: McdocType[] = []
 
 	for (const index of typeDef.parallelIndices) {
 		let lookup: string[] = []
 		if (index.kind === 'static') {
-			if (index.value === '%fallback') {
-				values = child.fields.filter(f => f.kind === 'pair').map(f => f.type)
-				break
```
Comment (misode) [inline packages/mcdoc/src/runtime/checker/index.ts:890]: Is this removal intended? Does `foo: minecraft:item[%fallback]` still work?

File: packages/java-edition/src/mcfunction/checker/index.ts [hunk 2]
```diff
@@ -217,13 +217,31 @@ const itemStack: core.SyncChecker<ItemStackNode> = (node, ctx) => {
...
-		groupedComponents.get(key)!.push(pair.key)
+		groupedComponents.get(componentId)!.push(pair.key)
 		if (pair.value) {
-			nbt.checker.index('minecraft:data_component', key)(pair.value, ctx)
+			if (componentId === 'minecraft:custom_data') {
+				if (pair.value.type === 'nbt:string') {
+					// TODO: Maybe move this to the nbt package
+					const stringNBT = nbt.parser.compound(
+						new core.Source(pair.value.value, pair.value.valueMap),
+						ctx,
```
Comment (misode) [inline packages/java-edition/src/mcfunction/checker/index.ts:231]: Yeah... it's not ideal that it has to run the parser here. The `#[nbt=mcdoc:custom_item_data[[%parent.id]]] string` mcdoc type should already be parsing the string as an NBT compound.

File: packages/mcdoc/src/runtime/checker/index.ts [hunk 3]
```diff
@@ -979,9 +975,17 @@ function simplifyIndexed<T>(
...
 		if (currentValues.includes(undefined)) {
-			// fallback case
-			values = child.fields.filter(f => f.kind === 'pair').map(f => f.type)
-			break
+			const fallbackDispatch = child.fields.find(f =>
+				f.kind === 'pair' && f.key.kind === 'literal' && f.key.value.value === '%fallback'
+			)
+
+			if (fallbackDispatch) {
+				values.push(fallbackDispatch.type)
```
Comment (misode) [inline packages/mcdoc/src/runtime/checker/index.ts:983]: What are these changes supposed to achieve? Why was this necessary?

File: packages/java-edition/src/mcfunction/checker/index.ts (hunk 2 above)
Comment (MulverineX) [reply packages/java-edition/src/mcfunction/checker/index.ts:231]: It does, but not in this context. Also, the `nbt` attribute doesn't actually check against the definition yet.

File: packages/mcdoc/src/runtime/checker/index.ts (hunk 3 above)
Comment (MulverineX) [reply packages/mcdoc/src/runtime/checker/index.ts:983]: Because otherwise, it creates a union of all dispatched types, rather than using the defined `%fallback`

File: packages/java-edition/src/mcfunction/checker/index.ts (hunk 2 above)
Comment (misode) [reply packages/java-edition/src/mcfunction/checker/index.ts:231]: The `nbt` attribute does run the checker...

File: packages/mcdoc/src/runtime/checker/index.ts (hunk 1 above)
Comment (MulverineX) [reply packages/mcdoc/src/runtime/checker/index.ts:890]: @NeunEinser would know, he said I should remove it

Comment (MulverineX) [reply packages/mcdoc/src/runtime/checker/index.ts:890]: (Answered in Discord)

File: packages/java-edition/src/mcfunction/checker/index.ts (hunk 2 above)
Comment (MulverineX) [reply packages/java-edition/src/mcfunction/checker/index.ts:231]: (its kinda bugged, there was discussion in discord)

**Review by misode — APPROVED** [review]
Seems to work well. The string parsing can be refactored in the future. I'll trust what neun was doing while pair-coding the fallback changes 👍
